src/core/server/ui_settings/ui_settings_client.ts

```diff
@@ -143,8 +146,17 @@ export class UiSettingsClient implements IUiSettingsClient {
   }
 
   private async getRaw(): Promise<UiSettingsRaw> {
+    const cachedValue = this.rawCache.get();
```

Comment:
I'm not really happy with 2 levels of caching, but I'd defer to the full refactoring of the `UI settings` service to address the problem.

Reply:
++ Agreed on this, I think the perf benefits make this worth it as an interim solution, but really we should restructure this during a proper refactoring